### PR TITLE
Add RKNN / Object Detection Pipeline

### PR DESCRIPTION
**Pull request by mdurrani808**

### CODE REVIEW

**Comment by mcm001** [comment]
Style wise I would vote for RKNN and DNN to be the same pipeline type, since all the tuning controls will be identical anyways. Thoughts?

File: photon-core/src/main/java/org/photonvision/common/hardware/Platform.java [hunk 1]
```diff
@@ -43,6 +45,13 @@ public enum Platform {
             true,
             OSType.LINUX,
             true), // Raspberry Pi 3/4 with a 64-bit image
+    LINUX_RK3588_64(
+        "Linux AARCH 64-bit with RK388",
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/common/hardware/Platform.java:49]: RK388? lol

File: photon-core/src/main/java/org/photonvision/common/hardware/Platform.java [hunk 2]
```diff
@@ -186,7 +198,12 @@ private static Platform getCurrentPlatform() {
                 return LINUX_32;
             } else if (RuntimeDetector.isArm64()) {
                 // TODO - os detection needed?
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/common/hardware/Platform.java:200]: lol

File: photon-core/src/main/java/org/photonvision/jni/RknnDetector.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    public static RknnDetector getRknnDetector() {
+        if (instance == null)
+            instance = new RknnDetector();
+ 
+        return instance;
+    }
+
+    public static synchronized void forceLoad() throws IOException {
+        TestUtils.loadLibraries();
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/jni/RknnDetector.java:36]: Smelly a bit

File: photon-core/src/main/java/org/photonvision/jni/RknnDetector.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        
+        forceLoad(getRknnDetector(), RknnDetector.class, List.of("rknn_jni", "rga", "rknnrt"));
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/jni/RknnDetector.java:38]: Order here seems sus

File: photon-core/src/main/java/org/photonvision/jni/RknnDetector.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            new Rect2d(it.left,it.top,it.right,it.bottom), it.class_id, it.conf
+        )).collect(Collectors.toList());
+    }
+    public static void release() {
+        if(!hasBeenDestroyed) {
+            RknnJNI.destroy(objPointer);
+            hasBeenDestroyed = true;
+        }
+        else {
+            System.out.print("aint no way bros tryna release the RKNN detector again skull");
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/jni/RknnDetector.java:54]: Cap

File: photon-core/src/main/java/org/photonvision/jni/RknnDetector.java (hunk 3 above)
Comment (mdurrani808) [reply photon-core/src/main/java/org/photonvision/jni/RknnDetector.java:35]: how so

**Comment by mdurrani808** [comment]
To do

- [ ] Fix this so the crosshair isn't in the top right and the id isn't the class id but 0 - n (someone else who knows how this stuff works more can fix this faster than i can)
- [ ] Make sure the user doesnt get option to make an RKNN pipeline on non orange pi platforms i will tackle this tommorow)
- [ ] Check what needs to be added to the orange pi image to get this to cook (i will tackle this tommorow)
- [ ] Do performance testing
- [ ] Add code to JNI to coremask to use all NPU cores (i will tackle this tommorow)
- [ ] add performance optimization scripts to the image 
- [ ] Add handles to the JNI for the parameters discussed on discord (there are all defined in post process.h) maybe @mcm001 or someone or i can take a shot at it if not enough bandwith 
- [ ] Add .jar loading properly @mcm001 
- [ ] Refactor this into an ML detector and then you can switch the backends
- [ ] Test this?

**Review by Bankst — APPROVED** [review]

**Review by mcm001 — APPROVED** [review]
Obama.png

**Pull request merged**